### PR TITLE
Minor fixes pre 2.27

### LINKED ISSUE #302: Documentation for log_inv_logit_diff is missing
#### Summary:
The function ``log_inv_logit_diff`` has been exposed to the Stan language, but its documentation is still missing from the Stan Functions Reference.

#### Description:
Documentation for ``log_inv_logit_diff`` is missing in the Stan Functions Reference [3.14 Composed Functions](https://mc-stan.org/docs/2_25/functions-reference/composed-functions.html). Since the function is vecorized, the documentation for it could look something like this:

>``R`` **``log_inv_logit_diff``** ``(T1 x, T2 y)``
natural logarithm of the difference of the inverse logit function of x and the inverse logit function of y.

#### Additional Information:
The LaTeX representation (in case it should be included as well):
\mathrm{log\_inv\_logit\_diff}(x,y) = \log(\mathrm{inv\_logit}(x) - \mathrm{inv\_logit}(y))

The function can be useful when one wants to code a customized version of an ordered logit model, e.g. an ordered multinomial logit model.

#### Current Version:
v2.25.0

### LINKED ISSUE #307: Add note to include .exe when checking the Stan compiler on Windows
#### Summary:
The instructions for checking the compiler in the CmdStan User's Guide, section 1.5 Checking the Stan compiler, does not work in Windows as `.exe` need to be added.

The User's Guide states:
```
make examples/bernoulli/bernoulli    # on Windows use mingw32-make
```
However to compile the model on Windows one must write:
```
mingw32-make examples/bernoulli/bernoulli.exe
```
Note that `.exe` is needed.

#### Current Output:
```
mingw32-make examples/bernoulli/bernoulli
mingw32-make: *** No rule to make target 'examples/bernoulli/bernoulli'.  Stop.
```

#### Expected Output:
```
mingw32-make examples/bernoulli/bernoulli.exe
--- Translating Stan model to C++ code ---
bin/stanc.exe  --o=examples/bernoulli/bernoulli.hpp examples/bernoulli/bernoulli.stan

--- Compiling, linking C++ code ---
g++ -std=c++1y -m64 -D_REENTRANT -Wall -Wno-unused-function -Wno-uninitialized -Wno-unused-but-set-variable -Wno-unused-variable -Wno-sign-compare -Wno-unused-local-typedefs -Wno-int-in-bool-context -Wno-attributes -Wno-ignored-attributes      -I stan/lib/stan_math/lib/tbb_2019_U8/include   -O3 -I src -I stan/src -I lib/rapidjson_1.1.0/ -I stan/lib/stan_math/ -I stan/lib/stan_math/lib/eigen_3.3.7 -I stan/lib/stan_math/lib/boost_1.72.0 -I stan/lib/stan_math/lib/sundials_5.2.0/include  -D_USE_MATH_DEFINES  -DBOOST_DISABLE_ASSERTS        -c  -x c++ -o examples/bernoulli/bernoulli.o examples/bernoulli/bernoulli.hpp
g++ -std=c++1y -m64 -D_REENTRANT -Wall -Wno-unused-function -Wno-uninitialized -Wno-unused-but-set-variable -Wno-unused-variable -Wno-sign-compare -Wno-unused-local-typedefs -Wno-int-in-bool-context -Wno-attributes -Wno-ignored-attributes      -I stan/lib/stan_math/lib/tbb_2019_U8/include   -O3 -I src -I stan/src -I lib/rapidjson_1.1.0/ -I stan/lib/stan_math/ -I stan/lib/stan_math/lib/eigen_3.3.7 -I stan/lib/stan_math/lib/boost_1.72.0 -I stan/lib/stan_math/lib/sundials_5.2.0/include  -D_USE_MATH_DEFINES  -DBOOST_DISABLE_ASSERTS              -Wl,-L,"C:/Users/user/cmdstan-2.25.0/stan/lib/stan_math/lib/tbb" -Wl,-rpath,"C:/Users/user/cmdstan-2.25.0/stan/lib/stan_math/lib/tbb"      examples/bernoulli/bernoulli.o src/cmdstan/main.o  -static-libgcc -static-libstdc++       stan/lib/stan_math/lib/sundials_5.2.0/lib/libsundials_nvecserial.a stan/lib/stan_math/lib/sundials_5.2.0/lib/libsundials_cvodes.a stan/lib/stan_math/lib/sundials_5.2.0/lib/libsundials_idas.a stan/lib/stan_math/lib/sundials_5.2.0/lib/libsundials_kinsol.a  stan/lib/stan_math/lib/tbb/tbb.dll -o examples/bernoulli/bernoulli.exe
rm -f examples/bernoulli/bernoulli.o
```

#### Additional Information:
I'm running from cmd on Windows 10.

#### Current Version:
v2.25.0

### LINKED ISSUE #342: Remove references to `is_infinite` should be `is_inf` in 18.1 of SUG
See the code in https://mc-stan.org/docs/2_26/stan-users-guide/truncated-random-number-generation.html

### LINKED ISSUE #348: `algebra_solver` is missing the data qualifier in function references and user guide
The algebra solver signature written here https://mc-stan.org/docs/2_26/functions-reference/functions-algebraic-solver.html and 
https://mc-stan.org/docs/2_26/stan-users-guide/coding-an-algebraic-system.html and possibly in other places is written as:

```
vector algebra_system(vector y, vector theta, real[] x_r, int[] x_i)
```
but should be
```
vector algebra_system(vector y, vector theta, data real[] x_r, data int[] x_i)
```

Stanc2 was not particularly strict with data qualifiers which could lead to C++ compiler problems (see https://github.com/stan-dev/stanc3/issues/866).

### PR DESCRIPTION
**Pull request by rok-cesnovar**
#### Summary

Closes a few more simple issues before 2.27:
- fixes #307 
- fixes #302 
- fixes #342 
- fixes #348 

#### Copyright and Licensing

Please list the copyright holder for the work you are submitting (this will be you or your assignee, such as a university or company):
Rok Češnovar

By submitting this pull request, the copyright holder is agreeing to license the submitted work under the following licenses:
- Code: BSD 3-clause (https://opensource.org/licenses/BSD-3-Clause)
- Documentation: CC-BY 4.0 (https://creativecommons.org/licenses/by/4.0/)

### CODE REVIEW

File: src/functions-reference/real-valued_basic_functions.Rmd [hunk 1]
```diff
@@ -1472,6 +1472,12 @@ of x and the natural exponentiation of y. \[
 `R` **`log_inv_logit`**`(T x)`<br>\newline
 natural logarithm of the inverse logit function of x
 
+<!-- R; log_inv_logit_diff; (T1 x, T2 y); -->
+\index{{\tt \bfseries log\_inv\_logit\_diff }!{\tt (T1 x, T2 y): R}|hyperpage}
+
+`R` **`log_inv_logit_diff`**`(T1 x, T2 y)`<br>\newline
+natural logarithm of the difference of the inverse logit function of x and the inverse logit function of y
```
Comment (bbbales2) [inline src/functions-reference/real-valued_basic_functions.Rmd:1479]: Ooof, I think we should have named this `log_diff_inv_logit` (log of difference of inverse logits) but I guess this is why the docs are needed

**Review by bbbales2 — APPROVED** [review]
Good

File: src/functions-reference/real-valued_basic_functions.Rmd (hunk 1 above)
Comment (rok-cesnovar) [reply src/functions-reference/real-valued_basic_functions.Rmd:1479]: Ah yeah.
